sqlite3_cache/cache.py

```diff
@@ -522,3 +525,99 @@ def ttl_many(self, keys: List[str]) -> Dict[str, int]:
             self._con.commit()
 
         return results
+
+    def _filter_key_result_list(self, unfiltered: List[Tuple[str, Any]]) -> List[str]:
+        """
+        Filters key result list to only those keys for cache items that are still alive,
+        and purges expired items from the cache
+        :param unfiltered: A list of key and expiration tuples
+        :return: A filtered list of keys
+        """
+        results: List[str] = []
+        to_delete: List[str] = []
+        for key, exp in unfiltered:
+            exp = self._exp_datetime(exp)  # noqa: PLW2901
+            if exp is not None and datetime.now(tz=timezone.utc) >= exp:
+                to_delete.append(key)
+                continue
+
+            results.append(key)
+
+        if to_delete:
+            self._con.execute(self._delete_many_sql.format(", ".join([f"'{key}'" for key in to_delete])))
+            self._con.commit()
+
+        return results
+
+    def get_all_keys(self) -> List[str]:
+        """
+        Get all keys that exist in the cache for currently valid cache items.
+
+        :return: List of cache keys in sort order.
+        """
+        fetched: List[Tuple[str, Any]] = self._con.execute(self._get_keys_sql).fetchall()
+
+        if not fetched:
+            return []
+
+        return self._filter_key_result_list(fetched)
+
+    def find_keys_starting_with(self, pattern: str) -> List[str]:
+        """
+        Find keys that start with the given pattern.
+        Matching follows the SQLite specification for the LIKE operator, so
+        it will match 'A' to 'a', but not 'Ä' to 'ä'.
+        Will only return keys that exist in the cache for currently valid cache items.
+
+        :param pattern: The pattern to match at the start of the key.
+        :return: List of matching cache keys in sort order.
+        """
+        data = {"pattern": f"{''}{pattern}%"}
```

Comment:
`{''}` seems unnecessary here.

Reply:
I jumped through hoops to reuse a SQL string and avoid linting complaints about a boolean flag, and then this was left behind from some refactoring. I can just make two SQL strings and all of this will become simpler.

Reply:
Fixed